process_hierarchy.py

```diff
@@ -0,0 +1,478 @@
+    cursor.execute("use hierarchy_db", None)
+    cursor.execute(
+        "insert into hierarchy_rec ( \
+            id, \
+            create_ts, \
+            type, \
+            name, \
+            status, \
+            display_name, \
+            parent_id) values \
+            ('1', now(), 'institution', 'unknown', 'Active', 'Unknown', null )\
+        )",
+        None,
+    )
+
+
+def delete_hierarchy_db(cursor):
+    """delete the database but only in initial development"""
+    if moc_db_helper_functions.db_exist(cursor, "hierarchy_db"):
+        cursor.execute("drop database hierarchy_db", None)
+
+
+def process_record(cursor, rec, current_dict):
+    """
+    processes the record, by checking if it currently exists
+
+    if it exists:
+        use it's current id
+        if it is modified add the modification
+    else
+        add the record
+
+    This way we can keep track of the history in an auditable
+    manner
+    """
+    hierarchy_id = find_hierarchy_id(rec["name"], current_dict)
+    if hierarchy_id:
+        add_update = False
+
+        if current_dict[hierarchy_id]["name"] != rec["name"]:
+            add_update = True
+        if current_dict[hierarchy_id]["display_name"] != rec["display_name"]:
+            add_update = True
+        if current_dict[hierarchy_id]["parent_id"] != rec["parent_id00"]:
+            add_update = True
+        if add_update:
+            # create a new record with a new id
+            cursor.execute(
+                "insert into hierarchy_db.hierarchy_rec ( \
+                id, \
+                create_ts, \
+                type, \
+                name, \
+                status, \
+                display_name, \
+                parent_id \
+                ) values ( \
+                %(id)s, \
+                now(), \
+                %(type)s, \
+                %(name)s, \
+                %(status)s, \
+                %(display_name)s, \
+                %(parent_id)s \
+                )",
+                {
+                    "id": hierarchy_id,
+                    "type": rec["type"],
+                    "name": rec["name"],
+                    "display_name": rec["display_name"],
+                    "status": rec["status"],
+                    "parent_id": rec["parent_id"],
+                },
+            )
+            # update the dictionary
+            current_dict[hierarchy_id]["type"] = rec["type"]
+            current_dict[hierarchy_id]["name"] = rec["name"]
+            current_dict[hierarchy_id]["display_name"] = rec["display_name"]
+            current_dict[hierarchy_id]["status"] = rec["status"]
+            current_dict[hierarchy_id]["parent_id"] = rec["parent_id"]
+            current_dict[hierarchy_id]["still_active"] = True
+    else:
+        # create a new record with a new id
+        cursor.execute(
+            "insert into hierarchy_db.hierarchy_rec ( \
```

Comment:
These two `insert` statements are broadly similar. Is there any way we can rewrite this function so that there's a single `insert` statement instead? That makes things a bit more maintainable because we won't need to make changes in two places whenever the schema changes.

Reply:
I believe so, this was easier to write.  I will come back to this time permitting.  If time is not permitting, will add an issue.

Reply:
I rewrote this to one insert statement.